File: 167. Two Sum II - Input array is sorted .java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        
        int left = 0;
        int right = numbers.length - 1;
        
        while(left < right){
            int sum = numbers[left] + numbers[right];
            if(sum < target){
                left++;
            } else if(sum > target) {
                right--;
            } else {
                return new int[]{left + 1, right + 1};
            }
        }
        return new int[0];
    }
}

File: 167. Two Sum II - Input array is sorted .go
func twoSum(numbers []int, target int) []int {
    left := 0
    right := len(numbers) - 1
    for left < right {
        sum := numbers[left] + numbers[right];
        if sum < target {
            left++;
        } else if sum > target {
            right--;
        } else {
            return []int{left + 1, right + 1}
        }
    }
    return []int{}
}
